src/reconcile/rec_visibility.c

```diff
@@ -644,7 +644,20 @@ __rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_UPDATE *first_upd
             }
         }
 
-        /* Always select the newest committed update to write to disk */
+        /*
+         * Don't write any update that is not stable if precise checkpoint is enabled.
+         *
+         * TODO: currently we only support this mode from startup. If we want to enable this through
+         * reconfiguration, we need to ensure we have run a rollback to stable before we run the
+         * first checkpoint with the precise mode.
+         */
+        if (F_ISSET(S2C(session), WT_CONN_PRECISE_CHECKPOINT) &&
```

Comment:
Performance nit, since update chains can be long, probably good to create a `conn` variable, so we're not calling `S2C` lots of times.